src/MSBuild/app.config

```
@@ -50,6 +50,7 @@
         <dependentAssembly>
           <assemblyIdentity name="XamlBuildTask" culture="neutral" publicKeyToken="31bf3856ad364e35" />
           <bindingRedirect oldVersion="4.0.0.0" newVersion="16.0.0.0" />
+          <bindingRedirect oldVersion="15.0.0.0" newVersion="16.0.0.0" />
```

Should we collapse this to a range, so we remember to update it again in the future?

```suggestion
          <bindingRedirect oldVersion="4.0.0.0-15.0.0.0" newVersion="16.0.0.0" />
```

? Or were there some intermediate ones that shouldn't be redirected?

---

@rainersigwald If I understand this correctly, the 4.0.0.0 applies to .NET framework version, and 15.0.0.0 implies VS version.  I am not sure what the intrinsic implication is if we change that into a range.   

On the newVesion side, do you have any suggestion about that range?  Could we use "16.0.0.0 - *"?  I am concerned that people will forget to update this with newer version of Visual Studio.

---

We also don't want to downgrade someone from, say, 17.x to 16.0.0.0. I'd at least extend it to 15.0.0.0-16.0.0.0, but I think it's better not to extend into the future.

I believe things will break in RPS if they rely on a new version of XamlBuildTask and ignore this, so as long as they don't push it in over RPS failures, I think we're ok.

---

`newVersion` can only have a single value, not a range. It needs to be the version that's currently shipping, so leaving it as 16.0.0.0 makes sense.

We prefer this, which will be easy to update and clearly shows "redirect existing versions to the current version". This isn't super duper relevant now, but will make a lot more sense when we bump to `17.0.0.0`.

```suggestion
          <bindingRedirect oldVersion="4.0.0.0-16.0.0.0" newVersion="16.0.0.0" />
```